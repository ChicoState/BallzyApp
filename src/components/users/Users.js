import React from 'react';
import ScrollableTabView, {DefaultTabBar} from 'react-native-scrollable-tab-view';
import Posts from './Posts';
import UserChallenges from './UserChallenges';

import {
  Text,
  View,
  StyleSheet,
  TouchableHighlight,
  Image,
  Button,
} from 'react-native';

import UsersStyles from '../../styles/UsersStyles'

class Users extends React.Component {
  static navigationOptions = {
    title: 'Users',
    tabBarIcon: () => (
      <Image
        source={require('../../img/users48x48.png')}
        style = {{width: 32, height: 32}}
      />
    ),
  };
  render() {
    return(
      <View style={{flex:10}}>
        <View style={{flex: 10}}>
          <View style={UsersStyles.UserInfo}>
            <View style={{flex: 3}}>
            </View>
            <View style={UsersStyles.profileArea}>
              <View style={{flex: 7}}>
                <Image
                  style={UsersStyles.image}
                  source={require('./profile.png')}
                />
              </View>
              <View style={{flex: 3}}>
                <View style={{flex: 1}}>
                  <Text>Followers:</Text>
                </View>
                <View style={{flex: 1}}>
                  <Text>Posts:</Text>
                </View>
              </View>
            </View>
            <View style={{flex: 3}}>
            </View>
          </View>
          <View style={UsersStyles.Content}>
            <ScrollableTabView
              renderTabBar={() => <DefaultTabBar />}
              tabBarBackgroundColor= '#E54747'
              tabBarInactiveTextColor = 'white'
              tabBarActiveTextColor = 'white'
              tabBarUnderlineStyle = {{
                backgroundColor: 'white',
                opacity: .2
              }}
            >
              <Posts tabLabel='Posts'/>
              <UserChallenges tabLabel='Challenges'/>
            </ScrollableTabView>
          </View>
        </View>
      </View>
    );
  }
}

export default Users;
