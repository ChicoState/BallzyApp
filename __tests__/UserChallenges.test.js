'use strict';

import 'react-native';
import React from 'react';
//import React from 'react';
import Users from '../src/components/users/UserChallenges';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Users />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});
