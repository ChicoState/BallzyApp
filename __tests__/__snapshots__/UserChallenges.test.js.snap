exports[`test renders correctly 1`] = `
<View>
  <View>
    <RCTScrollView>
      <View />
    </RCTScrollView>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    UserChallenges
  </Text>
</View>
`;
